File: src/app/entities/incident/incident.ts
import { Call } from "../call/call";
import {} from "../element/element"
import { Multimedia } from "../multimedia";
import { Resolution } from "../resolution";
import {Element} from "../element/element"

export class Incident {
  id : number = 0;
  type: string = '';
  priority: number = 0;
  address : string = '';
  confirmed : boolean = false;
  status: string = 'Drafted';
  eta: string = '';
  ata: string = '';
  timeOfIncident: string ='';
  etr: string = '';
  calls: number = 0;
  voltage : number = 0;
  affectedConsumers :number = 0;
  sheduledTime : string = '';
  creator? : string = '';
  elements : Element[] = [];
  call : Call[] = [];
  resolutions : Resolution[] = [];
  multimedia : Multimedia[] = [];
  //crew : any = [];
  //version : number = 0;


  constructor(type: string,priority: number,address : string,confirmed : boolean,
    status: string,
    eta: string,
    ata: string,
    timeOfIncident: string,
    etr: string,
    calls: number,
    voltage : number,
    affectedConsumers :number,
    sheduledTime : string ){
      this.type = type;
      this.priority =priority;
      this.address =address;
      this.confirmed = confirmed;
      this.status = status;
      this.eta = eta;
      this.ata = ata;
      this.timeOfIncident = timeOfIncident;
      this.etr = etr;
      this.calls = calls;
      this.voltage =  voltage;
      this.affectedConsumers =  affectedConsumers;
      this.sheduledTime =  sheduledTime;
    }

}

File: WEB2BEKEND/WEB2BEKEND/Models/Incident.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace WEB2BEKEND.Models
{
  public class Incident
  {
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public string Type { get; set; }
    public int Priority { get; set; }
    public bool Confirmed { get; set; }
    public string Status { get; set; }
    public string Eta { get; set; }
    public string Ata { get; set; }
    public string TimeOfIncident { get; set; }
    public string Etr { get; set; }
    public int Calls { get; set; }
    public int Voltage { get; set; }
    public int AffectedConsumers { get; set; }
    public string SheduledTime { get; set; }
    public string Creator { get; set; }
    public string Address { get; set; }

    public ICollection<IncidentElement> Elements { get; set; }
    public ICollection<IncidentCall> Call { get; set; }
    public ICollection<IncidentResolution> Resolutions { get; set; }
    public ICollection<Multimedia> Multimedia { get; set; }
  }
}
